graphyte.py

```diff
@@ -66,13 +72,19 @@ def stop(self):
 
     def build_message(self, metric, value, timestamp, tags={}):
         """Build a Graphite message to send and return it as a byte string."""
-        if not metric or metric.split(None, 1)[0] != metric:
+        if has_whitespace(metric):
             raise ValueError('"metric" must not have whitespace in it')
         if not isinstance(value, (int, float)):
             raise TypeError('"value" must be an int or a float, not a {}'.format(
                 type(value).__name__))
 
-        tags_suffix = ''.join(';{}={}'.format(x[0], x[1]) for x in sorted(tags.items()))
+        tags_suffix = ''
```

Comment:
Using str `+=` in Python can lead to O(N^2) behavior, so better to build a list and join. Hopefully no one has that many tags, but might as well use best practices. So maybe:

```
tags_strs = [';{}={}'.format(k, v) for k, v in sorted(tags.items())]
if any(has_whitespace(t) for t in tags_strs):
    raise ValueError('"tags" keys and values must not have whitespace in them')
tags_suffix = ''.join(tags_strs)
```

Reply:
Oh, didn't think of that!
I change it to this better approach anyway! Just made a change to cast to _unicode_. 
`tags_strs = [u';{}={}'.format(k, v) for k, v in sorted(tags.items())]`